src/hotspot/share/utilities/concurrentHashTable.inline.hpp

```diff
@@ -511,18 +512,20 @@ inline void ConcurrentHashTable<CONFIG, F>::
     // We left critical section but the bucket cannot be removed while we hold
     // the _resize_lock.
     bucket->lock();
-    size_t nd = delete_check_nodes(bucket, eval_f, BULK_DELETE_LIMIT, ndel);
+    GrowableArrayCHeap<Node*, F> extra(0);
+    size_t nd = delete_check_nodes(bucket, eval_f, BULK_DELETE_LIMIT, ndel_stack, extra);
     bucket->unlock();
     if (is_mt) {
       GlobalCounter::write_synchronize();
     } else {
       write_synchonize_on_visible_epoch(thread);
     }
     for (size_t node_it = 0; node_it < nd; node_it++) {
-      del_f(ndel[node_it]->value());
-      Node::destroy_node(_context, ndel[node_it]);
+      Node* ndel = node_it < BULK_DELETE_LIMIT ? ndel_stack[node_it] : extra.at(node_it - BULK_DELETE_LIMIT);
```

Comment:
Needs to be a reference (`Node*& ndel`) or `DEBUG_ONLY(ndel = (Node*)POISON_PTR;)` will be a no-op.

Reply:
Will fix, thanks!